### DISCUSSION TITLE
is it possible to read parquet files using clickhouse remotely ?

### DISCUSSION THREAD

**Discussion by dridk** [ClickHouse/ClickHouse#54657] [Q&A] [answered]
Hi, 

- I Have datas stored as parquet files on my server filesystem. ( NFS mount )
- I have a clickhouse instance on the same server. 
- from a desktop client ( dbeaver ) , I can connect to the clickhouse database

I wonder if I can query parquet file from the client ? Can we create something like Table engine or view ?

**Comment by den-crane** [comment, marked as answer]
you can use `file` function.

For example I have

```
ls -1 /mnt/nfs/
data_NONE.Parquet
```

Now I need to create a symlink, because CH is able to read only `/var/lib/clickhouse/user_files` by default (alternativly you can reconfigure it).

```
cd /var/lib/clickhouse/user_files
ln -s /mnt/nfs/ nfs
```

Now I can list files like

```
/var/lib/clickhouse/user_files# ls -1 nfs/
data_NONE.Parquet
```

Also be careful with NFS, it should be readable by a clickhouse user, use `sudo -u clickhouse` to check from clickhouse user.

```
sudo -u clickhouse ls -l nfs/
total 772
-rw-r--r-- 1 denis.zhuravlev WebAMG.StJohn 786436 Sep 14 18:27 data_NONE.Parquet
```

Now:

```
SELECT
    _path,
    _file,
    count(),
    max(bool)
FROM file('nfs/*.Parquet')
GROUP BY
    1,
    2

Query id: 6a542ae0-6140-44a6-9fc8-54b34892bfe8

┌─_path────────────────────────────────────────────────┬─_file─────────────┬─count()─┬─max(bool)─┐
│ /var/lib/clickhouse/user_files/nfs/data_NONE.Parquet │ data_NONE.Parquet │       3 │ true      │
└──────────────────────────────────────────────────────┴───────────────────┴─────────┴───────────┘
```

**Reply by dridk** [reply]
Thanks ! Looks easy

Does it mean that all clickhouse user has access to those files ? Is there a way to restrict access to a specific user ?

**Reply by den-crane** [reply]
>all clickhouse user has access to those files

it does. All who have execute `file` priv.

>Is there a way to restrict access to a specific user ?

No
